cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/FullModel.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.cgmes.model;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.exceptions.UncheckedXmlStreamException;
+import com.powsybl.commons.xml.XmlUtil;
+
+import javax.xml.XMLConstants;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.Reader;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class FullModel {
+
+    private final String id;
+
+    private final ZonedDateTime scenarioTime;
+
+    private final ZonedDateTime created;
+
+    private final String description;
+
+    private final int version;
+
+    private final List<String> profiles;
+
+    private final List<String> dependentOn;
+
+    private final List<String> supersedes;
+
+    private final String modelingAuthoritySet;
+
+    public FullModel(String id, ZonedDateTime scenarioTime, ZonedDateTime created, String description, int version, List<String> profiles,
+                     List<String> dependentOn, List<String> supersedes, String modelingAuthoritySet) {
+        this.id = Objects.requireNonNull(id);
+        this.scenarioTime = Objects.requireNonNull(scenarioTime);
+        this.created = Objects.requireNonNull(created);
+        this.description = description;
+        this.version = version;
+        this.profiles = Objects.requireNonNull(profiles);
+        this.dependentOn = Objects.requireNonNull(dependentOn);
+        this.supersedes = Objects.requireNonNull(supersedes);
+        this.modelingAuthoritySet = Objects.requireNonNull(modelingAuthoritySet);
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public ZonedDateTime getScenarioTime() {
+        return scenarioTime;
+    }
+
+    public ZonedDateTime getCreated() {
+        return created;
+    }
+
+    public Optional<String> getDescription() {
+        return Optional.ofNullable(description);
+    }
+
+    public int getVersion() {
+        return version;
+    }
+
+    public List<String> getProfiles() {
+        return profiles;
+    }
+
+    public List<String> getDependentOn() {
+        return dependentOn;
+    }
+
+    public List<String> getSupersedes() {
+        return supersedes;
+    }
```

Comment:
You should return immutable version of these lists

Reply:
As it is also passed to contrusctor we should maybe make a defensive copy in the constructor

Reply:
Done, defensive copy in the contructor

Reply:
It's not equivalent... How do you prevent a user to clear your own copy?
```
fullModel.getDependentOn().clear();
```
I think the copy in the constructor is not necessary because a new context is created each time the parsing is called, but you shoud protect your own data.